test/unit/domain/users-ips/ip-metadata-validator.spec.ts

```diff
@@ -0,0 +1,154 @@
+import {
+  InvalidIPMetadataASNError,
+  InvalidIPMetadataCountryError,
+  InvalidIPMetadataProxyError,
+  MissingIPMetadataError,
+} from "@domain/errors"
+import { IPMetadataValidator } from "@domain/users-ips/ip-metadata-validator"
+
+const defaultConfig = {
+  denyIPCountries: [],
+  allowIPCountries: [],
+  denyASNs: [],
+  allowASNs: [],
+}
+
+const defaultIpInfo: IPType = {
+  ip: "89.187.173.251" as IpAddress,
+  firstConnection: new Date(),
+  lastConnection: new Date(),
+  asn: "AS60068",
+  provider: "ISP",
+  country: "United States",
+  isoCode: "US",
+  region: "Florida",
+  city: "Miami",
+  proxy: false,
+}
+
+describe("IPMetadataValidator", () => {
+  describe("validateForReward", () => {
+    it("returns true for empty config", () => {
+      const validator =
+        IPMetadataValidator(defaultConfig).validateForReward(defaultIpInfo)
+      expect(validator).toBe(true)
+    })
+
+    it("returns true for an allowed country", () => {
```

Comment:
Can it return false/error for a country not in a defined allowed-list? Can probably add test-cases for:
- country not in defined allow-list fails
- country not in defined deny-list passes

Reply:
Added additional tests in regards to your suggestion https://github.com/GaloyMoney/galoy/pull/1232/commits/2507717d52130b03fb7521f8a35202c44460fb7c